pytorch_to_returnn/torch/nn/modules/linear.py

```diff
@@ -123,6 +122,41 @@ def create_returnn_layer_dict(self, *inputs: Tensor, **kwargs) -> Dict[str, Any]
     return {"class": "dot", "red1": red1, "red2": red2, "var1": var1, "var2": var2, "from": sources}
 
 
+class DotProductAlongDim(Module):
```

Comment:
I think I would simply call it `Dot` or `GenericDot`, and allow for potential generalization later when needed.